main.py

```diff
@@ -0,0 +1,115 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/")
+def root():
+    return {"message": "Welcome to Library API"}
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    skip: int = 0, limit: int = 10, db: Session = Depends(get_db)
+):
+    return crud.get_author_list(db=db, skip=skip, limit=limit)
+
+
+@app.get("/authors/{author_id}", response_model=schemas.Author)
+def read_author(author_id: int, db: Session = Depends(get_db)):
+    author = crud.get_author(db=db, author_id=author_id)
+
+    if not author:
+        raise HTTPException(status_code=404, detail="Author not found")
+
+    return author
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    return crud.create_author(db=db, author=author)
+
+
+@app.put("/authors/{author_id}", response_model=schemas.Author)
+def update_author(
+    author_id: int, author: schemas.AuthorCreate, db: Session = Depends(get_db)
+):
+    updated_author = (
+        crud.update_author(db=db, author_id=author_id, author=author)
+    )
+
+    if not updated_author:
+        raise HTTPException(status_code=404, detail="Author not found")
+
+    return updated_author
+
+
+@app.delete("/authors/{author_id}", response_model=dict)
+def delete_author(author_id: int, db: Session = Depends(get_db)):
+    deleted_successfully = crud.delete_author(db=db, author_id=author_id)
+    if deleted_successfully:
+        return {"message": "Author deleted successfully"}
+
+    raise HTTPException(status_code=404, detail="Author not found")
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(
+    author_id: int | None = None,
```

Comment:
The type hint `int | None` is used here. While this is valid in Python 3.10 and later, if you are using an earlier version, you should use `Optional[int]` from the `typing` module instead.